massa-execution-worker/src/context.rs

```diff
@@ -575,24 +596,46 @@ impl ExecutionContext {
     /// * `to_addr`: optional crediting address (use None for pure coin destruction)
     /// * `amount`: amount of coins to transfer
     /// * `check_rights`: check that the sender has the right to spend the coins according to the call stack
+    /// * `current_slot`: necessary for check vesting (use None if from_addr = None)
     pub fn transfer_coins(
         &mut self,
         from_addr: Option<Address>,
         to_addr: Option<Address>,
         amount: Amount,
         check_rights: bool,
+        current_slot: Option<Slot>,
     ) -> Result<(), ExecutionError> {
-        // check access rights
-        if check_rights {
-            if let Some(from_addr) = &from_addr {
-                if !self.has_write_rights_on(from_addr) {
-                    return Err(ExecutionError::RuntimeError(format!(
-                        "spending from address {} is not allowed in this context",
-                        from_addr
-                    )));
+        if let Some(from_addr) = &from_addr {
+            // check access rights
+            if check_rights && !self.has_write_rights_on(from_addr) {
+                return Err(ExecutionError::RuntimeError(format!(
+                    "spending from address {} is not allowed in this context",
+                    from_addr
+                )));
+            }
+
+            if let Some(current_slot) = current_slot {
+                // control vesting min_balance for sender address
+                if let Some(vesting_range) = VestingRange::find_vesting_range(
```

Comment:
For me `find_vesting_range()` shouldn't be in `VestingRange` as it don't use any of the member of the structure and doesn't do anything related to the structure. It should be on the type `PreHashMap<Addresss, VestingRange>` but if you don't do a special type for it you can put it in the context.

Reply:
Ok then you can just do a function inside of anything private in this file.